packages/contrast-colors/utils.js

```diff
@@ -0,0 +1,506 @@
+      if (letter === 's' || letter === 'l' || letter === 'v') {
+        colorObject[letter] = round(ch, 2);
+        if (!object) {
+          rnd = round(ch * 100);
+          rnd += '%';
+        }
+      } else if (letter === 'h' && !object) {
+        rnd += 'deg';
+      }
+    }
+    return rnd;
+  });
+
+  const stringName = space;
+  const stringValue = `${stringName}(${newColorObj.join(', ')})`;
+
+  if (object) {
+    return colorObject;
+  }
+  return stringValue;
+}
+
+function luminance(r, g, b) {
+  const a = [r, g, b].map((v) => {
+    v /= 255;
+    return v <= 0.03928 ? v / 12.92 : ((v + 0.055) / 1.055) ** 2.4;
+  });
+  return (a[0] * 0.2126) + (a[1] * 0.7152) + (a[2] * 0.0722);
+}
+
+function getContrast(color, base, baseV) {
+  if (baseV === undefined) { // If base is an array and baseV undefined
+    const baseLightness = chroma.rgb(...base).hsluv()[2];
+    baseV = round(baseLightness / 100, 2);
+  }
+
+  const colorLum = luminance(color[0], color[1], color[2]);
+  const baseLum = luminance(base[0], base[1], base[2]);
+
+  const cr1 = (colorLum + 0.05) / (baseLum + 0.05); // will return value >=1 if color is darker than background
+  const cr2 = (baseLum + 0.05) / (colorLum + 0.05); // will return value >=1 if color is lighter than background
+
+  if (baseV < 0.5) { // Dark themes
+    // If color is darker than background, return cr1 which will be whole number
+    if (cr1 >= 1) {
+      return cr1;
+    }
+    // If color is lighter than background, return cr2 as negative whole number
+    return -cr2;
+  }
+  // Light themes
+  // If color is lighter than background, return cr2 which will be whole number
+  if (cr1 < 1) {
+    return cr2;
+  }
+  // If color is darker than background, return cr1 as negative whole number
+  if (cr1 === 1) {
+    return cr1;
+  }
+  return -cr1;
+}
+
+function minPositive(r) {
+  if (!r) { throw new Error('Array undefined'); }
+  if (!Array.isArray(r)) { throw new Error('Passed object is not an array'); }
+  return Math.min(...r.filter((val) => val >= 1));
+}
+
+function ratioName(r) {
+  if (!r) { throw new Error('Ratios undefined'); }
+  r = r.sort((a, b) => a - b); // sort ratio array in case unordered
+
+  const min = minPositive(r);
+  const minIndex = r.indexOf(min);
+  const nArr = []; // names array
+
+  const rNeg = r.slice(0, minIndex);
+  const rPos = r.slice(minIndex, r.length);
+
+  // Name the negative values
+  for (let i = 0; i < rNeg.length; i++) {
+    const d = 1 / (rNeg.length + 1);
+    const m = d * 100;
+    const nVal = m * (i + 1);
+    nArr.push(round(nVal));
+  }
+  // Name the positive values
+  for (let i = 0; i < rPos.length; i++) {
+    nArr.push((i + 1) * 100);
+  }
+  nArr.sort((a, b) => a - b); // just for safe measure
+
+  return nArr;
+}
+
+// Binary search to find index of contrast ratio that is input
+// Modified from https://medium.com/hackernoon/programming-with-js-binary-search-aaf86cef9cb3
+function getMatchingRatioIndex(list, value) {
```

Comment:
Looks like this is no longer in use and is replaced with the Search Color function?

Reply:
You are right. I think it's safe to remove it. Will do in next commit.